### PR TITLE
Implement BCSQ parsing and a few minor fixes.

### PR DESCRIPTION
**Pull request by cassimons**
Hi Danial,

I came across your vcf to json converter and it was just want I needed.

I found I needed to add BCSQ parsing and I also ran into a couple of small issues. I am not sure if you would prefer this as a single pull request, or if you would like it broken up.

These commits include:

- Building out the BCSQ parsing stub you had in the code using similar logic to ANN. I needed to add an csq annotated vcf file to the test data for this.
- Added conversion for NaN floats to json null.
- Fixed the logic for missing int to null conversion. In my hands the int.low comparison needed to be explicitly set to int32.low.
- Fixed how the GT format field is handled when ALL is specified.
- Prevented the trailing comma being printed after the last variant when using --array. This was causing a parsing error with strict downstream consumer (eg Chrome).

I think I have added tests covering each of these.  If you would like more details on the issues that I was running into please let me know.

Happy to change things if there are aspects you would prefer to handle differently. Just let me know.

Thanks again.

Cheers

### CODE REVIEW

**Comment by brentp** [comment]
would be nice to have a "consequence" or similar module for these types of things. I have a half-baked thing in slivar that handles snpEff, CSQ, BCSQ: https://github.com/brentp/slivar/blob/master/src/slivarpkg/tsv.nim#L83

**Comment by danielecook** [comment]
Hey sorry I was out of town and am just seeing this! Thanks for your contributions.

**Comment by danielecook** [comment]
> would be nice to have a "consequence" or similar module for these types of things. I have a half-baked thing in slivar that handles snpEff, CSQ, BCSQ: https://github.com/brentp/slivar/blob/master/src/slivarpkg/tsv.nim#L83

@brentp looks useful I'll see if I can add something similar.

File: src/vcf2json.nim [hunk 1]
```diff
@@ -20,10 +21,21 @@ proc `%`(s: int): JsonNode =
   # Overload JsonNode from int
   # Important for converting
   # '.' to null values
-  if s == int.low:
+  ## When a format field has multiple values per sample, missing values are represeted by 
+  ## int32.low OR (int32.low +1). I can not understand this at all?
```
Comment (danielecook) [inline src/vcf2json.nim:25]: I'm not sure about the int32.low + 1 but I am guessing this code works?

**Comment by danielecook** [comment]
@epistatic only comment would be to merge into dev instead of master in the future. Thanks!

File: src/vcf2json.nim (hunk 1 above)
Comment (cassimons) [reply src/vcf2json.nim:25]: It works... but it is not exactly pretty. After implementing this I opened an hts-nim issue to ask what the expected behaviour was (https://github.com/brentp/hts-nim/issues/57). Thanks to Brent, I now understand the problem and can see a more general way of addressing it.

I will give it a go at using something along the linens of what Brent suggested to clean this up and also handle missing values for types that are not explicitly handled at the moment.

Comment (cassimons) [reply src/vcf2json.nim:25]: Will use dev in the future (sorry).  
Cheers.

**Pull request merged**